_posts/2020-05-30-terraform.md

```diff
@@ -0,0 +1,138 @@
+1. Sync OpenShift service account credentials into Jenkins credential store
+1. Roll-out Jenkins to add the new credentials by starting a Jenkins job executing an Ansible playbook
+1. Create reverse proxy configuration in reverse proxy repository
+1. Use a self-service portal to request access for Jenkins to update the configuration on the reverse proxy
+1. Roll-out the reverse proxy configuration using a Jenkins job
+1. Roll-out the environment configuration and application landscape to the new namespace through a Jenkins job that runs OC process and OC applies using the configuration.
+
+This sums up the best-case scenario.
+Since it requires multiple teams to perform disconnected tasks at different times, errors are frequent and slow to resolve.
+Next to that, there is a lot of manual copy and paste involved to sync configuration from different sources into different repositories and systems.
+Since any manual action, especially involving copying configuration between locations, is prone to errors, this introduces even more failure points.
+Combining the time it takes for tickets to be executed, copying the manual configuration to the correct locations and the debugging involved in getting the environment online results in lead times expressed in weeks.
+On one occasion the lead time for a single environment for a new team was as high as 6 weeks.
+This is one of the reasons that at least 1 or 2 spare environments are provisioned in order to provide teams with at least a minimal environment to start working with.
+This of course introduces additional cost.
+
+This is just one of the scenarios which need to be supported.
+Since the choice was made to migrate away from the on-premise solution and onto the public cloud, it was an ideal time to automate these scenarios as well.
+
+## Why use Terraform to supercharge infrastructure provisioning
+
+As mentioned, one of the main drivers to look into automation for infrastructure was the move to the public cloud.
+The reason why Azure was selected, might be discussed in a future blog post.
+
+As Azure was selected, the logical choice for automating the infrastructure provisioning would be ARM templates.
+An attempt was made at setting up the infrastructure with Terraform together with an ARM expert.
+Unfortunately, multiple walls were hit due to the nature of the corporate setup on Azure.
+One of the limitations that were hit, was the inability for ARM to make changes to resources that were provided by a central team.
+More specifically, using a VNET that wasn't managed by ARM to deploy an Azure Kubernetes Service proved to be difficult or even impossible.
+
+Using the CLI and scripts to create the setup was attempted as well, but this didn't fit the vision of a declarative setup for provisioning infrastructure.
+Re-applying the same scripts either broke the setup or created additional, unneeded resources.
+A clear no-go.
+
+As a third option, Terraform was investigated.
```

Comment:
What is the difference here exactly compared to the first option? Seeing as Terraform was also investigated there.
I assume this option is about solely using Terraform?

Reply:
Added